### DISCUSSION TITLE
Email Mangling

### DISCUSSION THREAD

**Discussion by siefkenj** [protz/thunderbird-conversations#1574] [Q&A]
It currently looks like Conversations scans for blockquotes and signatures to and autohides them if they are large. But the code that does this seems ad hoc. Is there a overall vision of what it's supposed to do?

A related question: is there any way to get a large dump of emails that don't include private information so that they could be used for tests, etc.? Maybe the Thunderbird tests themselves contain a bunch of sample emails that could be reused for Conversations tests?

**Comment by Standard8** [comment]
> It currently looks like Conversations scans for blockquotes and signatures to and autohides them if they are large. But the code that does this seems ad hoc. Is there a overall vision of what it's supposed to do?

I think the main idea is that if it is over 4/5 lines, then it will hide the block. @protz might be able to describe more.

Also, there's probably things like mailman mailing lists where we shouldn't do the collapsing if possible.

> A related question: is there any way to get a large dump of emails that don't include private information so that they could be used for tests, etc.? Maybe the Thunderbird tests themselves contain a bunch of sample emails that could be reused for Conversations tests?

The only one I've heard of was not very representative, so it probably wouldn't be worth it.

**Reply by protz** [reply]
That's the main idea. The code has always been very ad-hoc and was based on a set of test emails I had gathered in a private folder, that were more or less representative of what I had been able to find back when I initially wrote the addon.

A lot of the stuff is to work around the misbehaved clients, such as lotus notes, or outlook, or outlook web access, which all ignore RFC-style `>`-based quoting.

The heuristic of collapsing anything longer than 4-5 lines worked well, because either someone was top- or bottom-replying, and then the whole email gets collapsed, which is good, since it's probably already in another message in the conversation. Or, if someone is inline-replying, then it's likely that only small snippets are quoted and these don't get collapsed . That's the idea...

**Reply by siefkenj** [reply]
Thanks for the story @protz .  Do you happen to still have those email samples? Could you anonymize them so they could be incorporated into tests?
